packages/experiments/src/components/Toggle/Toggle.state.ts

```diff
@@ -32,12 +36,15 @@ export class ToggleState extends BaseState<IToggleProps, IToggleViewProps, ITogg
   };
 
   private _onClick = (ev: React.MouseEvent<HTMLElement>) => {
-    const { disabled, onChange } = this.props;
+    const { label, disabled, onChange } = this.props;
     const { checked } = this.state;
 
     if (!disabled) {
+      let ariaLabel: string = typeof label === 'string' ? label : 'No label';
+      ariaLabel += !checked ? ': On' : ': Off';
+
       // Only update the state if the user hasn't provided it.
-      this.setState({ checked: !checked });
+      this.setState({ ariaLabel: ariaLabel, checked: !checked });
```

Comment:
Does `ariaLabel` have to be stored in state? It seems like it's just a transform function on `label` and `checked`.